server/util.go

```diff
@@ -254,7 +254,7 @@ func dumpBinaryRow(buffer []byte, columns []*ColumnInfo, row types.Row) ([]byte,
 		case mysql.TypeDouble:
 			buffer = dumpUint64(buffer, math.Float64bits(row.GetFloat64(i)))
 		case mysql.TypeNewDecimal:
-			buffer = dumpLengthEncodedString(buffer, hack.Slice(row.GetMyDecimal(i).String()))
+			buffer = dumpLengthEncodedString(buffer, hack.Slice(string(row.GetMyDecimal(i).ToString())))
```

Comment:
Why change this?

Reply:
mmm...In MySQL `select -0.0000000000000000000000000000000000000000000000000017382578996420603` will not round up to zero...so I try to run test case ahead in here to see what happen if we don't truncate it- -